### PR TITLE
Add Strategy for local authed user 

### PR DESCRIPTION
**Pull request by fraxachun**
Depends on https://github.com/vivid-planet/comet/pull/896

### CODE REVIEW

File: demo/admin/src/dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -36,20 +38,28 @@ const Greeting = styled(Typography)`
     color: white;
 `;
 
+const meQuery = gql`
+    query Me {
```
Comment (nsams) [inline demo/admin/src/dashboard/Dashboard.tsx:42]: Das query muss eigentlich eindeutig benannt sein, weil du importierst ja GQLMeQuery aus einem zentralen graphql.generated. Wenn du exakt das gleiche query 2x brauchst musst du es irgendwo exportieren - was ich aber nicht machen würde.

File: demo/api/src/auth/current-user.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { CurrentUser as CometCurrentUser } from "@comet/cms-api";
+import { Field, ObjectType } from "@nestjs/graphql";
+
+@ObjectType()
+export class CurrentUser extends CometCurrentUser {
```
[the remark this answers is not in the document]
Comment (fraxachun) [reply demo/api/src/auth/current-user.ts:5]: The class is needed for the GraphQL-API. I think, this solution is okay, inheriting is easy. Furthermore I think using a concrete class for type definitions is better than module augmentation.

File: packages/api/cms-api/src/auth/strategies/authed-user.strategy.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { forwardRef, Inject, Injectable } from "@nestjs/common";
+import { PassportStrategy } from "@nestjs/passport";
+import { Strategy } from "passport-custom";
+
+import { AUTH_CONFIG } from "../auth.constants";
+import { AuthedUserConfig } from "../auth.module";
+
+@Injectable()
+export class AuthedUserStrategy<CurrentUser> extends PassportStrategy(Strategy, "authedUser") {
```
[the remark this answers is not in the document]
Comment (fraxachun) [reply packages/api/cms-api/src/auth/strategies/authed-user.strategy.ts:9]: I'd say "AuthedUser" should be sufficient, if more votes go for a more specific name my choice would be "StaticAuthedUserStrategy".

File: demo/admin/src/dashboard/Dashboard.tsx (hunk 1 above)
Comment (fraxachun) [reply demo/admin/src/dashboard/Dashboard.tsx]: It's now currentUser and currentUserSignOut. In my opinion more logical and specific enough.

File: demo/api/src/auth/current-user.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (nsams) [reply demo/api/src/auth/current-user.ts:5]: It has one major problem:

- you need to use generics to inject the application specific CurrentUser everywhere in every module/resolver/whatever, starting always in application code (as that has only knowledge of the class to inject)
- that is often not really possible (how would you do that for https://github.com/vivid-planet/comet/blob/main/packages/api/cms-api/src/auth/decorators/get-current-user.decorator.ts?)

File: demo/admin/src/dashboard/Dashboard.tsx (hunk 1 above)
Comment (nsams) [reply demo/admin/src/dashboard/Dashboard.tsx]: You did misunderstand that :) (me is perfectly ok and used by many apis)

`query **Me** {` This part as to be unique, you need eg.

```
query DashboardCurrentUser {
    me {
    }
}
```

File: demo/api/src/auth/current-user.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (fraxachun) [reply demo/api/src/auth/current-user.ts:5]: Okay, valid point, but even though getCurrentUser is not correctly typed it is possible to type it with the CurrentUser-Object of the application when using it in a resolver. In contrary I don't know how to create a typed resolver in Comet without using generics. Do you have a suggestion based on an interface with module augmentation?

File: demo/admin/src/dashboard/Dashboard.tsx (hunk 1 above)
Comment (fraxachun) [reply demo/admin/src/dashboard/Dashboard.tsx]: I changed it to DashboardCurrentUser. However, I still like CurrentUser more than Me.

File: demo/api/src/app.module.ts [hunk 4]
```diff
@@ -66,23 +64,22 @@ import { RedirectScope } from "./redirects/dto/redirect-scope";
...
-                config: {
-                    apiPassword: config.API_PASSWORD,
-                    idpConfig: {
-                        url: config.IDP_API_URL,
-                        password: config.IDP_API_PASSWORD,
-                        clientId: config.IDP_CLIENT_ID,
-                    },
+            strategy: "authedUser",
+            useFactory: () => ({
+                authedUser: {
```
Comment (johnnyomair) [inline demo/api/src/app.module.ts:71]: Is this only used for local development? If not, how would an application get the current user?

Comment (fraxachun) [reply demo/api/src/app.module.ts:71]: Mainly for local development, but it is also possible to use a static user in a deployed environment which is secured by other means. Authorization for production environments is using JWTs now: https://github.com/vivid-planet/comet/pull/903

File: packages/api/cms-api/src/auth/strategies/authed-user.strategy.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (fraxachun) [reply packages/api/cms-api/src/auth/strategies/authed-user.strategy.ts:9]: I renamed it to StaticAuthedUser in https://github.com/vivid-planet/comet/pull/903

File: demo/api/src/auth/current-user.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (nsams) [reply demo/api/src/auth/current-user.ts:5]: > it is possible to type it with the CurrentUser-Object

of course, but that is inconvenient and easy to forget.

> Do you have a suggestion based on an interface with module augmentation?

probably both approaches combined:

- library defines an interface that the application extends with module augmentation
- application defines class (that implements the interface)

The class needs only to be injected where you need to instanciate it, everywhere else you can work with the interface

[the remark this answers is not in the document]
Comment (fraxachun) [reply demo/api/src/auth/current-user.ts:5]: Changed in https://github.com/vivid-planet/comet/pull/903

**Comment by fraxachun** [comment]
Due to some bigger changes I'm closing this PR and all reviews should go to https://github.com/vivid-planet/comet/pull/903 now

**Pull request closed without merging**